operator/k8s_cep_gc.go

```diff
@@ -70,6 +71,7 @@ func enableCiliumEndpointSyncGC() {
 					podsCache[pod.Namespace+"/"+pod.Name] = &pod
```

Comment:
It worries me a little bit that we are doing an entire Pod List of the cluster every N minutes. This function should be re-written once https://github.com/cilium/cilium/pull/11195 is merged.

Reply:
Is your worry that we do this every 10min instead of 30min now? We have always been doing this so far. This only changes the interval.

We can definitely optimize this. I'm not sure a watcher is always better though. it will pay the cost of churn. Whereas right now we pay the snapshot cost of all pods which can be a lot less or a lot more. It will vary.

Reply:
I wasn't aware about this listing at all, that's why I made the comment. The watcher will be good because we can optimize it and only keep the pod name and namespace in memory (instead of all fields of the pod). Can be optimized as a follow up.